core/domain/improvements_domain.py

```diff
@@ -0,0 +1,184 @@
+            target_type, target_id, issue_description, status, resolver_id,
+            resolved_on):
+        self.entity_type = entity_type
+        self.entity_id = entity_id
+        self.entity_version = entity_version
+        self.task_type = task_type
+        self.target_type = target_type
+        self.target_id = target_id
+        self.issue_description = issue_description
+        self.status = status
+        self.resolver_id = resolver_id
+        self.resolved_on = resolved_on
+
+    @property
+    def task_id(self):
+        """Returns the unique identifier of this task.
+
+        Returns:
+            str. The ID of this task.
+        """
+        return improvements_models.TaskEntryModel.generate_task_id(
+            self.entity_type, self.entity_id, self.entity_version,
+            self.task_type, self.target_type, self.target_id)
+
+    @property
+    def composite_entity_id(self):
+        """Returns value of helper field for identifying the task's target
+        entity.
+
+        Returns:
+            str. The value of the helper field.
+        """
+        return improvements_models.TaskEntryModel.generate_composite_entity_id(
+            self.entity_type, self.entity_id, self.entity_version)
+
+    @classmethod
+    def from_payload_dict(
+            cls, payload_dict, resolver_id=None, resolved_on=None):
+        """Returns a new TaskEntry domain object from the given payload dict and
+        optional resolver ID.
+
+        Args:
+            payload_dict: dict. Contains the following keys:
+                entity_type: str. The type of entity the task entry refers to.
+                    For example, "exploration".
+                entity_id: str. The ID of the entity the task entry refers to.
+                    For example, an exploration ID.
+                entity_version: int. The version of the entity the task entry
+                    refers to. For example, an exploration's version.
+                task_type: str. The type of task the task entry tracks.
+                target_type: str. The type of sub-entity the task entry refers
+                    to. For example, "state" when entity type is "exploration".
+                target_id: str. The ID of the sub-entity the task entry refers
+                    to. For example, the state name of an exploration.
+                issue_description: str. The sentence generated by Oppia to
+                    describe why the task was created.
+                status: str. Tracks the state/progress of the task entry.
+            resolver_id: str or None. The corresponding user who resolved this
+                task. Only used when payload dict's status is resolved.
+            resolved_on: datetime or None. The datetime at which this task was
+                resolved. Only used when payload dict's status is resolved.
+
+        Returns:
+            improvements_domain.TaskEntry.
```

Comment:
Please always include a description with your docstrings, not just a typeinfo. That's the current codebase convention.